File: 4.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

enum Constants { SIZE_ARRAY = 1000 };

int main() {
    int *array = (int*) malloc(SIZE_ARRAY * sizeof(int));
    long long count = 0;
    while(scanf("%d", &array[count]) != EOF) {
        count++;
        if(count >= SIZE_ARRAY) {
            long long unsigned int size = (count / SIZE_ARRAY) * SIZE_ARRAY;
            array = (int*)realloc(array, size * sizeof(int));
        }
    }
	if(count < 2) {
        return 0;
    }
    long long min_day = 0, max_day = 1, delta = array[1] - array[0];
    for(long long i = 0; i < count - 1; i++) {
        for(long long j = i + 1; j < count; j++) {
            long long tmp_delta = array[j] - array[i];
            if(tmp_delta > delta) {
                delta = tmp_delta;
                min_day = i;
                max_day = j;
            } else if(tmp_delta == delta) {
                long long delta_day = j - i;
                long long delta_day_prew = max_day - min_day;
                if(delta_day < delta_day_prew) {
                    delta = tmp_delta;
                    min_day = i;
                    max_day = j;
                } else if(max_day > j && delta_day == delta_day_prew) {
                    delta = tmp_delta;
                    min_day = i;
                    max_day = j;
                }
            }
        }
    }
    printf("%d %d %d\n", delta, min_day, max_day);
    fflush(stdout);
}

File: 4.py
temp_array = input().split()

min_day = 0
max_day = 0
delta = 0
for i in range(len(temp_array)):
    for j in range(i + 1, len(temp_array)):
        a = int(temp_array[j])
        b = int(temp_array[i])
        tmp_delta = a - b
        if(tmp_delta > int(delta)):
            delta = int(tmp_delta)
            min_day = (i)
            max_day = (j)
        elif tmp_delta == int(delta):
            delta_day = j - i
            a = int(max_day)
            b = int(min_day)
            delta_day_prew = a - b
            if(delta_day < delta_day_prew):
                delta = int(tmp_delta)
                min_day = (i)
                max_day = (j)
print(delta, min_day, max_day)